ipn/store.go

```diff
@@ -64,3 +64,14 @@ func ReadStoreInt(store StateStore, id StateKey) (int64, error) {
 func PutStoreInt(store StateStore, id StateKey, val int64) error {
 	return store.WriteState(id, fmt.Appendf(nil, "%d", val))
 }
+
+// HandleTCPStateKeyForDomain returns a StateKey that stores the
+// whitespace-separated list of TCP ports that tailscaled should handle for the
+// given domain name (such as "foo.donkey-fish.ts.net")
+//
+// The returned value from the state store might be "443 8080", for example.
+// How the port is handled by tailscaled is subject to other StateKey
+// lookups. TODO(bradfitz): document those, once they're fleshed out.
+func HandleTCPStateKeyForDomain(domain string) StateKey {
```

Comment:
Can you put this in prefs instead? I would rather we don't add more global keys with FUS.

Reply:
It's qualified by the TCD name. I think it'd be pretty gross in prefs.

Reply:
Updated to JSON with single key in StateStore, per call